instrumentation/mongo/mongo-async-3.3/javaagent/build.gradle.kts

```diff
@@ -17,6 +17,11 @@ dependencies {
 
   library("org.mongodb:mongodb-driver-async:3.3.0")
 
+  // Starting with 3.7.0, mongodb-driver-async depends on mongodb-driver-core which contains
+  // both sync (com.mongodb.MongoClientSettings) and async (com.mongodb.async.SingleResultCallback)
+  // classes, causing the mongo-3.7 instrumentation to match instead of mongo-async-3.3.
+  latestDepTestLibrary("org.mongodb:mongodb-driver-async:3.6.+")
```

Comment:
The reason this matters in relation to this PR is because we assert the duration metric by scope, and all the telemetry emitted from this module when using testLatestDeps have the mongo-3.7 scope instead of the async one. 

Alternatively I could handle this by having conditional logic about the scope name in the the test assertion if we would prefer to not restrict latestDeps